framework/web/Request.php

```diff
@@ -162,7 +162,39 @@ class Request extends \yii\base\Request
      * @see getBodyParams()
      */
     public $parsers = [];
+    /**
+     * @var array The configuration for trusting security related header.
+     * The array key is a regular expression for matching the hostname.
+     * The value is a list of headers to trust.
+     * Example 1: trust all headers from domains ending in '.trusted.com'
+     * Example 2: trust just the x-forwarded-for header from domains ending in '.partial.com'
+     * ```
+     * [
+     *     '/^.*\.trusted.com$/',
+     *     '/^.*\.partial.com$/' => ['X-Forwarded-For']
+     *
+     * ]
+     * ```
+     * The default value trusts all headers from all hosts (to not break BC).
+     */
+    public $trustedHostConfig = [
+        '//'
+    ];
```

Comment:
We could use simplified pattern, like in https://github.com/yiisoft/yii2/blob/35998449eba8e295f5ed420350765c1196bf13e3/framework/filters/HostControl.php#L68-L91